src/aposteriori/dnn/network/visualization/prediction_entropy.py

```diff
@@ -11,70 +11,87 @@
 ) -> ampal.assembly:
     """
-    Assigns a B-factor to each residue equivalent to the prediction entropy
-    of the model.
+    Assigns a data point to each residue equivalent to the prediction the
+    tag value.
 
     Parameters
     ----------
     ampal_structure : ampal.Assembly or ampal.AmpalContainer
         Ampal structure to be modified. If an ampal.AmpalContainer is passed,
         this will take the first Assembly in the ampal.AmpalContainer `ampal_structure[0]`.
-    prediction_entropy : numpy.ndarray of floats
-        Numpy array with entropy on predictions (n,) where n is the number
-        of residues in the structure.
+    data_to_annotate : numpy.ndarray of numpy.ndarray of floats
+        Numpy array with data points to annotate (x, n) where x is the
+        numer of arrays with data points (eg, [ entropy, accuracy ] ,
+        x = 2n) and n is the number of residues in the structure.
+    tags : t.List[str]
+        List of string tags of the pdb object (eg. "b-factor")
 
     Returns
     -------
     ampal_structure : Assembly
         Ampal structure with modified B-factor values.
     """
+    assert len(tags) == len(
+        data_to_annotate
+    ), "The number of tags to annotate and the type of data to annotate have different lengths."
     # Deals with structures from NMR as ampal returns Container of Assemblies
     if isinstance(ampal_structure, ampal.AmpalContainer):
         warnings.warn(
             f"Selecting the first state from the NMR structure {ampal_structure.id}"
         )
         ampal_structure = ampal_structure[0]
 
-    # Reset B-factor:
-    for atom in ampal_structure.get_atoms(ligands=True, inc_alt_states=True):
-        atom.tags["bfactor"] = 0
-
-    total_length = 0
-    # Apply entropy as B-Factor to each chain
-    for i, chain in enumerate(ampal_structure):
-        # Check if chain is Polypeptide (it might be DNA for example...)
-        if isinstance(chain, Polypeptide):
-            total_length += len(chain)
-            for residue, entropy_val in zip(
-                chain, prediction_entropy[total_length : len(chain)]
-            ):
-                for atom in residue:
-                    atom.tags["bfactor"] = entropy_val
-    # Check whether the residues in chains were all covered by a posteriori
-    assert total_length == len(prediction_entropy), (
-        f"Expected a prediction for each residue, but chain is "
-        f"{len(total_length)} and entropy is {len(prediction_entropy)}"
-    )
+    if len(data_to_annotate) > 1:
+        assert len(data_to_annotate[0]) == len(data_to_annotate[1]), (
+            f"Data to annotatate has shape {len(data_to_annotate[0])} and "
+            f"{len(data_to_annotate[1])}. They should be the same."
+        )
+
+    for i, tag in enumerate(tags):
+        # Reset existing values:
+        for atom in ampal_structure.get_atoms(ligands=True, inc_alt_states=True):
+            atom.tags[tag] = np.min(data_to_annotate[i])
```

Comment:
Could you add a note to the doc string that states that the bfactor is reset to the minimum value?

Reply:
Yes, I should probably also add an explanation as to why we do that instead of setting it to zero.